sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -234,7 +234,7 @@ class SparkSession private(
   /**
    * A collection of methods for registering user-defined data sources.
    */
-  private[sql] def dataSource: DataSourceRegistration = sessionState.dataSourceRegistration
```

Comment:
Could you add `@since` tag because we open this, @allisonwang-db ?

Reply:
Also, please set appropriate tags like the following.
```
@Experimental
@Unstable
```

Reply:
oh yeah

Reply:
Will do!